### DISCUSSION TITLE
How to specify scope?

### DISCUSSION THREAD

**Discussion by SineStriker** [xmake-io/xmake#3051] [Q&A] [answered]
CMake allows to specify the scope so that an option's visiblity varies with each target.
```cmake
target_compile_definitions(MY_LIB PRIVATE PRIVATE_MACRO)
target_compile_definitions(MY_LIB PUBLIC PUBLIC_MACRO)

target_link_libraries(MY_EXE PRIVATE MY_LIB)
````
I don't know how to explictly specify the `PRIVATE` or `PUBLIC` scope when using `add_defines` or `add_links` in xmake.

**Comment by waruqi** [comment, marked as answer]
add_defines("xxx", {public = true})

文档里有说明，找下
